discord/commands/context.py

```diff
@@ -83,85 +78,74 @@ class ApplicationContext(discord.abc.Messageable):
     command: :class:`.ApplicationCommand`
         The command that this context belongs to.
     """
+    command: Optional[ApplicationCommand]
+
+    def __init__(
+        self,
+        bot: Bot,
+        interaction: Interaction,
+        *,
+        command: Optional[ApplicationCommand] = None,
+        args: List[Any] = None,
+        kwargs: Dict[str, Any] = None,
+        **kwargs2
+    ):
+        super().__init__(bot=bot, command=command, args=args, kwargs=kwargs, **kwargs2)
 
-    def __init__(self, bot: Bot, interaction: Interaction):
-        self.bot = bot
         self.interaction = interaction
 
         # below attributes will be set after initialization
-        self.command: ApplicationCommand = None  # type: ignore
         self.focused: Option = None  # type: ignore
         self.value: str = None  # type: ignore
         self.options: dict = None  # type: ignore
 
-        self._state: ConnectionState = self.interaction._state
-
-    async def _get_channel(self) -> Optional[InteractionChannel]:
-        return self.interaction.channel
+    async def reinvoke(self, *, call_hooks: bool = False, restart: bool = True) -> None:
```

Comment:
`reinvoke` is not a good name for a function that bypasses checks, cooldowns, and error handlers. `invoke_raw` is more descriptive.

Reply:
I personally think reinvoke makes sense especially in the 'context' of
```py
@command()
async def test(ctx):
    try:
        # error
    except:
        await ctx.reinvoke()
```
which would probably be the most likely the use case

Reply:
IDK if I'm just stupid, but that example doesn't make sense to me

Reply:
Looking at the example a little more, it does make sense to me now

Reply:
Just to clarify, what I meant was that it would most likely be called within the command callback. Outside the command, a "raw invoke" would probably be done by calling the command object directly (`Command(...)(ctx, *args, **kwargs)`)